smartsim/wlm/slurm.py

```diff
@@ -235,22 +236,23 @@ def _get_alloc_cmd(nodes, time, account, options=None):
     if account:
```

Comment:
I think making the function `set_account` not need to have an account value be passed sounds *extremely* unintuitive for any API user, and ends up doing nothing (so why even call it). IMO, the null default value would be a poor choice. 

A better option might be to change the default from None to empty string on `get_allocation`. It's simple and avoids the `x or default` hacks to appease the type checker.

You might have identified a secondary issue, too. The `if account:` lines look like guards against setting the value to some "empty value." This would seem to leave us with undefined behavior - if a  user is trying to clear that value (or changes it with a bad input), it would retain the old value and unexpectedly work with the wrong credentials, leading to erratic failure!

### Extra Notes

- I'd personally love to get rid of the `resources` dictionary as the underlying storage so the values can be easily typed. I would guess the intent was to be flexible in what values you can store as a setting, but the dictionary keys are all hardcoded so that is an illusion. 

- I'd also like to get rid of this `set_xxx` naming style for methods and have them be actual properties...

- I'd also like to get rid of the random `which('salloc')` here since we run that validation for every single command in `def detect_launcher() -> str:`.

Reply:
Per convo, updating to None for clarity/consistency.